hax/hax/util.py

```diff
@@ -239,6 +239,13 @@ def get_local_nodename(self) -> str:
             raise HAConsistencyException('Failed to communicate '
                                          'to Consul Agent') from e
 
+    def _get_service_health(self, service_name: str) -> List[Dict[str, Any]]:
```

Comment:
do we need  @repeat_if_fails() here?

Reply:
[10:06 AM] Konstantin Nekrasov
    No, we don't need it there. The call is used in _execute method. There is a logic to handle HAConsistencyException
​[10:06 AM] Konstantin Nekrasov
    If we add @repeat_if_fails, then we'll just add unnecessary sleeps.
​[10:07 AM] Mandar Sawant
    okay.

Reply:
No, we don't need it here. The call is effectively used by [_execute](https://github.com/knekrasov/cortx-hare/blob/fix-servicemon-exit-on-consul-failure/hax/hax/servicemon.py#L80) method. If we add `@repeat_if_fails`, we'll just introduce unnecessary sleeps.

`@repeat_if_fails()` can be considered (very roughly) as a border of transaction. If we have a solid business method - then we probably need it. If we're implementing a kind of small piece of functionality that will be used by a greater block of business logic - we don't need it. Instead we just need to ensure that in case of intermittent Consul exception HAConsistencyException is raised so that the caller can process it either with `@repeat_if_fails` or with a custom try-catch.